test-common/utils.c

```diff
@@ -2967,13 +2967,21 @@ eos_test_client_prepare_volume (EosTestClient *client,
                                                                       "..",
                                                                       "libeos-updater-util",
                                                                       NULL);
+  gchar *ld_library_path = g_getenv ("LD_LIBRARY_PATH");
+  g_autofree gchar *new_ld_library_path = NULL;
+  if (ld_library_path == NULL || *ld_library_path == '\0')
+    new_ld_library_path = g_strdup (libeos_updater_util_path);
+  else
+    new_ld_library_path = g_strdup_printf ("%s:%s", libeos_updater_util_path, ld_library_path);
+
   g_autoptr(GFile) sysroot = get_sysroot_for_client (client->root);
   CmdEnvVar envv[] =
     {
       { "EOS_UPDATER_TEST_UPDATER_DEPLOYMENT_FALLBACK", "yes", NULL },
       { "OSTREE_SYSROOT", NULL, sysroot },
       { "OSTREE_SYSROOT_DEBUG", "mutable-deployments", NULL },
       { "GI_TYPELIB_PATH", libeos_updater_util_path, NULL },
+      { "LD_LIBRARY_PATH", new_ld_library_path, NULL },
```

Comment:
I suspect you might need the `.libs` subdirectory, since that is where libtool puts the `.so`, but let’s see whether the tests pass.